### PR TITLE
onError link

### PR DESCRIPTION
**Pull request by elitan**
Fixes https://github.com/nhost/react-apollo/issues/5

### CODE REVIEW

**Comment by elitan** [comment]
@qpre What do you think about this PR?

File: dist/index.d.ts [hunk 1]
```diff
@@ -9,7 +9,8 @@ export function generateApolloClient(
   },
   publicRole?: string,
   cache?: any,
-  connectToDevTools?: boolean
+  connectToDevTools?: boolean,
+  onError: () => void
```
Comment (elitan) [inline dist/index.d.ts:13]: Is this `onError` definition correct @komninoschat ?

File: src/index.js [hunk 2]
```diff
@@ -85,9 +87,23 @@ export function generateApolloClient({
...
+    // console.log("inside onError");
+
+    // if (graphQLErrors)
+    //   graphQLErrors.forEach(({ message, locations, path }) =>
+    //     console.log(
+    //       `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`
+    //     )
+    //   );
+
+    // if (networkError) console.log(`[Network error]: ${networkError}`);
```
Comment (elitan) [inline src/index.js:101]: Remove before merge

File: src/index.js [hunk 3]
```diff
@@ -85,9 +87,23 @@ export function generateApolloClient({
...
+  });
+
   const client = new ApolloClient({
     ssr: ssr,
-    link: from([link]),
+    link: from([errorLink, link]),
```
Comment (elitan) [inline src/index.js:106]: If `onError` is not set we should try to avoid adding an empty `errorLink`

Comment (elitan) [reply src/index.js]: fixed

**Comment by qpre** [comment]
Hey @elitan xorry for the late answer, this seems about right 👍  thanks

**Pull request merged**